### ISSUE TITLE
AppState: storing cursor, returning array

### ISSUE THREAD

**Issue by steph643** [worona/meteorflux#17]
The doc states that you can store minimongo cursors in the AppState object. But from what I understand of the code, this is not true: you simply convert cursors to arrays using `fetch()`.

This has been debated before, but it seems cursors have performance benefits over arrays (see [here](https://forums.meteor.com/t/current-state-of-find-vs-fetch-in-templates/8039)). So it is usually discouraged to use `fetch()`, at least when rendering with Blaze.

I think it would be better to forget about cursors and let people use `fecth()` manually if they really want to:
```javascript
// save some state from the database
AppState.set('videoList', function() {
  return Videos.find({}).fetch();
});
```

**Comment by LuisHerranz** [comment]
I'm sorry but I don't get what's the issue here.

Is there something which doesn't work or doesn't work as expected?

**Comment by steph643** [comment]
Sorry, I wasn't clear at all. And also, I mixed up the "cursor" case and the "function returning a cursor" case. 

Working code:
```javascript
Posts = new Mongo.Collection(null);
let myCursor = Posts.find();
AppState.set('myCursor', myCursor);
myCursor = AppState.get('myCursor');
console.log(myCursor.count());
```
Output, as expected:
```
0
```
Non-working code:
```javascript
Posts = new Mongo.Collection(null);
AppState.set('myCursor', function() {
  return Posts.find();
});
const myCursor = AppState.get('myCursor');
console.log(myCursor.count());
```
Unexpected output:
```
Uncaught TypeError: myCursor.count is not a function
```

**Comment by LuisHerranz** [comment]
AppState is not designed to retrieve cursors, but an array instead.

This is by design, to mimic Blaze helpers and make AppState work consistently in both worlds.

If you do this:
```html
{{#each posts.items}}
  {{> post
{{/posts.items}}
```
you get an array even though posts.items is a cursor.

If you do this:
```javascript
_.each(AppState.get('posts.items'), item => {
  doSomething(item);
});
```
you get an array, like in Blaze, even though posts.items is a cursor.

I suggest you to store the count in a separate property:
```javascript
AppState.set('myCursor', function() {
  return Posts.find();
});
AppState.set('myCursor.count', () => Posts.find().count());
```

**Comment by steph643** [comment]
Using `count()` was just an example, to show that in the same situation your package sometimes returns cursors, sometimes arrays.
I have to admit I don't understand your explanation, but this is ok for now :-) You probably thought about it more than I did.
At this stage, all I want to say is this: Blaze is one of the Meteor rendering layouts and has not much to do with the business logic field, where this package belongs.

**Comment by LuisHerranz** [comment]
Our design approach here is that both `AppState.get('something')` and its Blaze helper `{{something}}` should retrieve exactly the same thing. That's the reason `AppState.get` will retrieve an array even if you store a Mongo Cursor.

**Comment by steph643** [comment]
But why are you saying that Blaze converts cursors to arrays? I am not aware of such conversion. See the following code:
```html
<body>
  {{#with getCursor}}
    {{>myTemplate}}
  {{/with}}
</body>

<template name="myTemplate">
</template>
```
```javascript
//client only code
Players = new Mongo.Collection(null);

Template.body.helpers({
  getCursor: function () { return Players.find(); }
});

Template.myTemplate.created = function() {
  console.log(this.data);
}
```
Output:
```
L…n.Cursor {collection: LocalCollection, sorter: null, matcher: M…o.Matcher, _selectorId: undefined, skip: undefined…}
```
See? A cursor remains a cursor.

**Comment by LuisHerranz** [comment]
With Blaze I mean [Spacebars](https://github.com/meteor/meteor/blob/devel/packages/spacebars/README.md).

**Comment by steph643** [comment]
Do you mean Spacebars converts cursors to arrays? Do you have an example?

**Comment by LuisHerranz** [comment]
Both `{{#each cursor}}` and `{{#each array}}` return an `Array`.

**Comment by steph643** [comment]
I really don't get it.

**Comment by luisherranz** [comment]
We have deprecated this functionality so I will close this now.
